### PR TITLE
[Feature] Add clip backbone

### PR DESCRIPTION
**Pull request by Scarecrow0**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification
Add the CLIP backbone for inference time  of Imagenet classification.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/clip/README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+| vit_base_patch16_clip_384.laion2b_ft_in12k_in1k |     86.57 |    49.37 |     87.17 |     98.02 | vit-base-p16_pt-64xb64_in1k-384 | vit_base_patch16_clip_384.laion2b_ft_in12k_in1k.bin |
+| vit_base_patch16_clip_224.openai_ft_in12k_in1k  |     86.57 |    16.86 |     85.99 |     97.72 | vit-base-p16_pt-64xb64_in1k-224 | vit_base_patch16_clip_224.openai_ft_in12k_in1k.bin  |
+| vit_base_patch32_clip_224.laion2b_ft_in1k       |     88.22 |     4.36 |     82.46 |     96.12 | vit-base-p32_pt-64xb64_in1k-224 | vit_base_patch32_clip_224.laion2b_ft_in1k.bin       |
+
+*Models with * are converted from the [official repo](https://github.com/rwightman/pytorch-image-models). The config files of these models are only for inference. We don't ensure these config files' training accuracy and welcome you to contribute your reproduction results.*
+
+## Citation
+
+```bibtex
+@misc{https://doi.org/10.48550/arxiv.2103.00020,
```
Comment (tonysy) [inline configs/clip/README.md:45]: @InProceedings{pmlr-v139-radford21a,
  title = 	 {Learning Transferable Visual Models From Natural Language Supervision},
  author =       {Radford, Alec and Kim, Jong Wook and Hallacy, Chris and Ramesh, Aditya and Goh, Gabriel and Agarwal, Sandhini and Sastry, Girish and Askell, Amanda and Mishkin, Pamela and Clark, Jack and Krueger, Gretchen and Sutskever, Ilya},
  booktitle = 	 {Proceedings of the 38th International Conference on Machine Learning},
  year = 	 {2021},
  series = 	 {Proceedings of Machine Learning Research},
  publisher =    {PMLR},
}

**Comment by tonysy** [comment]
The `configs/clip/vit_base_224_dataproc.py` and other similar files should be moved to `configs/_base_/datasets`, please follow the filename convention like `imagenet_bs64_clip_224.py`

File: configs/clip/vit_base_224_dataproc.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# _base_ = ['./pipelines/rand_aug.py']
```
Comment (tonysy) [inline configs/clip/vit_base_224_dataproc.py:1]:
```suggestion
```

File: configs/clip/vit_base_384_dataproc.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# _base_ = ['./pipelines/rand_aug.py']
```
Comment (tonysy) [inline configs/clip/vit_base_384_dataproc.py:1]:
```suggestion
```

File: configs/clip/vit_base_448_dataproc.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
+# _base_ = ['./pipelines/rand_aug.py']
```
Comment (tonysy) [inline configs/clip/vit_base_448_dataproc.py:1]:
```suggestion
```

File: mmcls/models/backbones/vision_transformer.py [hunk 5]
```diff
@@ -9,7 +9,7 @@
 from mmengine.model import BaseModule, ModuleList
 from mmengine.model.weight_init import trunc_normal_
 
-from mmcls.registry import MODELS
+from ..builder import MODELS
```
Comment (tonysy) [inline mmcls/models/backbones/vision_transformer.py:12]:
```suggestion
from mmcls.registry import MODELS
```
Please explain the reason of the modification?

**Review by tonysy — CHANGES_REQUESTED** [review]
As inline comments

File: configs/clip/metafile.yml [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+        - Multi-Head Attention
+        - Scaled Dot-Product Attention
+        - Tanh Activation
+    Paper:
+      URL: https://arxiv.org/abs/2201.09792
+      Title: Learning Transferable Visual Models From Natural Language Supervision
+    README: configs/clip/README.md
+
+Models:
+  - Name: vit_base_patch16_clip_384.openai_ft_in12k_in1k
```
Comment (mzr1996) [inline configs/clip/metafile.yml:20]: Please rename the model name according to https://mmclassification.readthedocs.io/en/dev-1.x/advanced_guides/convention.html
For example:
`clip-vit-base-p16_openai-pre_3rdparty_in1k`
`clip-vit-base-p16_laion2b-in12k-pre_3rdparty_in1k-384px`

**Comment by Scarecrow0** [comment]
> As inline comments

Changes are applied.

File: configs/clip/metafile.yml [hunk 7]
```diff
@@ -0,0 +1,255 @@
+Collection:
```
Comment (tonysy) [inline configs/clip/metafile.yml:1]:
```suggestion
Collections:
```

**Comment by Scarecrow0** [comment]
Request for further review. @mzr1996 

The directory of converted checkpoints for CLIP model: /mnt/petrelfs/share_data/lirongjie/clip_backbone_ckpt

File: configs/clip/metafile.yml [hunk 8]
```diff
@@ -0,0 +1,317 @@
...
+          Top 1 Accuracy: 86.87
+          Top 5 Accuracy: 98.05
+        Task: Image Classification
+    Weights: /mnt/petrelfs/share_data/lirongjie/clip_backbone_ckpt/vit_base_patch16_clip_384.openai_ft_in12k_in1k.bin
+    Config: configs/clip/vit-base-p16_pt-64xb64_in1k-384.py
+    Converted From:
+      Weights: https://huggingface.co/timm/vit_base_patch16_clip_384.openai_ft_in12k_in1k
+      Code: https://github.com/rwightman/pytorch-image-models
+
+  - Name: clip-vit-base-p16_openai_3rdparty_ft_in12k_in1k-224px
```
Comment (mzr1996) [inline configs/clip/metafile.yml:85]: Here, the name should be `clip-vit-base-p16_openai-in12k-pre_3rdparty_in1k`
1. Both OpenAI and in12k are used for pre-training, which should be placed at the second section.
2. `ft` is unnecessary.
3. `224px` is usually default setting, and we only specify it when it's not 224px.

File: configs/clip/metafile.yml [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+    Metadata:
+      FLOPs: 16855600128.0
```
Comment (mzr1996) [inline configs/clip/metafile.yml:87]: Flops should be an int number.

File: configs/clip/metafile.yml [hunk 10]
```diff
@@ -0,0 +1,317 @@
...
+    Results:
+      - Dataset:
+        - ImageNet-1k
+        - ImageNet-21k
+        Metrics:
+          Top 1 Accuracy: 85.99
+          Top 5 Accuracy: 97.72
+        Task: Image Classification
+    Weights: /mnt/petrelfs/share_data/lirongjie/clip_backbone_ckpt/vit_base_patch16_clip_224.openai_ft_in12k_in1k.bin
+    Config: configs/clip/vit-base-p16_pt-64xb64_in1k-224.py
```
Comment (mzr1996) [inline configs/clip/metafile.yml:102]: The config name should be `vit-base-p16_64xb64_in1k.py`. You don't need `pt` to specify a config for pre-train or fine-tuning by now.
And if the image size is not 224px, please use name like `vit-base-p16_64xb64_in1k-384px.py`

**Review by mzr1996 — CHANGES_REQUESTED** [review]

File: configs/clip/metafile.yml [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+      Version: v1.0.0
+
+Models:
+  - Name: clip-vit-base-p16_laion2b_3rdparty_ft_in1k-384px
+    Metadata:
+      FLOPs: 49370078208.0
+      Parameters: 86568424
+      Training Data:
+        - ImageNet-1k
+    In Collections: CLIP
```
Comment (mzr1996) [inline configs/clip/metafile.yml:29]: `In Collection`, instead of `In Collections`.

**Review by mzr1996 — CHANGES_REQUESTED** [review]

**Review by tonysy — CHANGES_REQUESTED** [review]
Please upadte the files according to zerun's comments

**Review by mzr1996 — APPROVED** [review]
